### ISSUE TITLE
如何以LORA的方式微调Mini-InternVL-Chat-2B-V1-5？

### ISSUE THREAD

**Issue by dengnianchen** [OpenGVLab/InternVL#302]
我注意到shell目录下有用于InternVL-Chat-V1-2-Plus的lora微调脚本，按照里面的参数去微调Mini-InternVL-Chat-2B-V1-5的模型时会报错，看起来是模型结构变动所致。

请问如果我想以LORA的方式微调Mini-InternVL-Chat-2B-V1-5该怎么做？

**Comment by bingwork** [comment]
+1

**Comment by L-Teer** [comment]
同问

**Comment by Single430** [comment]
+1

**Comment by syj2908** [comment]
+1

**Comment by wintercat1994** [comment]
+1

**Comment by wintercat1994** [comment]
https://github.com/OpenGVLab/InternVL/issues/240 我看到这个问题有提到用stage3的设置做2B模型的lora finetune 不知道是否有更多人尝试

**Comment by shiva-vardhineedi** [comment]
set -x

PARTITION=${PARTITION:-"INTERN2"}
GPUS=${GPUS:-2}  # Set the number of GPUs to 2
GPUS_PER_NODE=${GPUS_PER_NODE:-2}
QUOTA_TYPE=${QUOTA_TYPE:-"reserved"}
NODES=$((GPUS / GPUS_PER_NODE))
CPUS_PER_TASK=${CPUS_PER_TASK:-10}
SRUN_ARGS=${SRUN_ARGS:-""}
BATCH_SIZE=${BATCH_SIZE:-16}  # Further reduce total batch size
PER_DEVICE_BATCH_SIZE=${PER_DEVICE_BATCH_SIZE:-1}  # Reduce per-device batch size
GRADIENT_ACC=${GRADIENT_ACC:-4} 



export PYTHONPATH="${PYTHONPATH}:$(pwd)"
export MASTER_PORT=34229
export TF_CPP_MIN_LOG_LEVEL=3

OUTPUT_DIR='work_dirs/internvl_chat_v1_5_internlm2_1_8b_dynamic_res_finetune_new'

if [ ! -d "$OUTPUT_DIR" ]; then
  mkdir -p "$OUTPUT_DIR"
fi

# number of gpus: 2
# batch size per gpu: 1
# gradient accumulation steps: 4
# total batch size: 16
# epoch: 1
python -m torch.distributed.launch --nproc_per_node=${GPUS} --use_env internvl/train/internvl_chat_finetune.py \
  --model_name_or_path "Mini-InternVL-Chat-2B-V1-5" \
  --conv_style "internlm2-chat" \
  --output_dir ${OUTPUT_DIR} \
  --meta_path "shell/data/internvl_1_2_finetune.json" \
  --overwrite_output_dir True \
  --force_image_size 448 \
  --max_dynamic_patch 12 \
  --down_sample_ratio 0.5 \
  --drop_path_rate 0.1 \
  --pad2square False \
  --freeze_llm True \
  --freeze_mlp True \
  --freeze_backbone True \
  --use_llm_lora 16 \
  --vision_select_layer -1 \
  --use_data_resampling False \
  --dataloader_num_workers 4 \
  --bf16 True \
  --num_train_epochs 1 \
  --per_device_train_batch_size ${PER_DEVICE_BATCH_SIZE} \
  --gradient_accumulation_steps ${GRADIENT_ACC} \
  --evaluation_strategy "no" \
  --save_strategy "steps" \
  --save_steps 200 \
  --save_total_limit 3 \
  --learning_rate 4e-5 \
  --weight_decay 0.01 \
  --warmup_ratio 0.03 \
  --lr_scheduler_type "cosine" \
  --logging_steps 1 \
  --max_seq_length 8192 \
  --do_train True \
  --grad_checkpoint True \
  --group_by_length True \
  --dynamic_image_size True \
  --use_thumbnail True \
  --ps_version 'v2' \
  --deepspeed "zero_stage2_config.json" \
  --report_to "tensorboard" \
  2>&1 | tee -a "${OUTPUT_DIR}/training_log.txt"
  
  
  This worked for me... i have 2 A10Gs

**Comment by feihuamantian** [comment]
https://github.com/OpenGVLab/InternVL/issues/351

**Comment by czczup** [comment]
可以试一下更新到最新的代码，然后按照下面的文档来微调：

微调InternVL 2.0: https://internvl.readthedocs.io/en/latest/internvl2.0/finetune.html
微调InternVL 1.5: https://internvl.readthedocs.io/en/latest/internvl1.5/finetune.html
